dds_web/utils.py

```diff
@@ -30,6 +32,24 @@
 ####################################################################################################
 
 
+def contains_uppercase(input):
+    """Verify that string contains at least one upper case letter."""
+    if not re.search("[A-Z]", input):
+        raise marshmallow.ValidationError("Required: at least one upper case letter.")
+
+
+def contains_lowercase(input):
+    """Verify that string contains at least one lower case letter."""
+    if not re.search("[a-z]", input):
+        raise marshmallow.ValidationError("Required: at least one lower case letter.")
+
+
+def contains_digit_or_specialchar(input):
+    """Verify that string contains at least one special character OR digit."""
+    if not any(re.search(x, input) for x in ["[0-9]", "[#?!@$%^&*-]"]):
+        raise marshmallow.ValidationError("Required: at least one digit OR a special character.")
```

Comment:
Might want to report the set of special characters in the error message since there might be other characters that people would consider "special"?